internal/user/repository/eventsourcing/eventstore.go

```diff
@@ -691,6 +699,103 @@ func (es *UserEventstore) PasswordCodeSent(ctx context.Context, userID string) e
 	return nil
 }
 
+func (es *UserEventstore) AddExternalIDP(ctx context.Context, externalIDP *usr_model.ExternalIDP) (*usr_model.ExternalIDP, error) {
+	if externalIDP == nil || !externalIDP.IsValid() {
+		return nil, errors.ThrowPreconditionFailed(nil, "EVENT-Ek9s", "Errors.User.ExternalIDP.Invalid")
+	}
+	existingUser, err := es.UserByID(ctx, externalIDP.AggregateID)
+	if err != nil {
+		return nil, err
+	}
+	if existingUser.Human == nil {
+		return nil, errors.ThrowPreconditionFailed(nil, "EVENT-Cnk8s", "Errors.User.NotHuman")
+	}
+	repoUser := model.UserFromModel(existingUser)
+	repoExternalIDP := model.ExternalIDPFromModel(externalIDP)
+	aggregates, err := ExternalIDPAddedAggregate(ctx, es.Eventstore.AggregateCreator(), repoUser, repoExternalIDP)
+	if err != nil {
+		return nil, err
+	}
+	err = es_sdk.PushAggregates(ctx, es.PushAggregates, repoUser.AppendEvents, aggregates...)
+	if err != nil {
+		return nil, err
+	}
+
+	es.userCache.cacheUser(repoUser)
+	if _, idp := model.GetExternalIDP(repoUser.ExternalIDPs, externalIDP.UserID); idp != nil {
+		return model.ExternalIDPToModel(idp), nil
+	}
+	return nil, errors.ThrowInternal(nil, "EVENT-Msi9d", "Errors.Internal")
+}
+
+func (es *UserEventstore) BulkAddExternalIDPs(ctx context.Context, externalIDPs []*usr_model.ExternalIDP) error {
+	if externalIDPs == nil || len(externalIDPs) == 0 {
+		return errors.ThrowPreconditionFailed(nil, "EVENT-Ek9s", "Errors.User.ExternalIDP.MinimumExternalIDPNeeded")
+	}
+	for _, externalIDP := range externalIDPs {
+		if !externalIDP.IsValid() {
+			return caos_errs.ThrowPreconditionFailed(nil, "EVENT-idue3", "Errors.User.ExternalIDP.Invalid")
+		}
+	}
+	existingUser, err := es.UserByID(ctx, externalIDPs[0].AggregateID)
```

Comment:
using the id on first index might lead to problems as the function name `BulkAddExternalIDPs` does not make it clear that it must only be use on a single Aggregate